nixos/modules/services/databases/redis.nix

```diff
@@ -277,11 +283,18 @@ in
 
       serviceConfig = {
         ExecStart = "${cfg.package}/bin/redis-server /run/redis/redis.conf";
-        RuntimeDirectory = "redis";
-        StateDirectory = "redis";
         Type = "notify";
+        # User and group
         User = "redis";
         Group = "redis";
+        # Runtime directory and mode
+        RuntimeDirectory = "redis";
+        RuntimeDirectoryMode = "0750";
+        # State directory and mode
+        StateDirectory = "redis";
+        StateDirectoryMode = "0750";
```

Comment:
What is this needed for?

Reply:
Need to simple configure service `mastodon` and other services that can use the unix socket:
```
      (lib.mkIf cfg.redis.enableUnixSocket {
        unixSocket = "/run/redis/redis.sock";
        unixSocketPerm = 770;
      })
```

Reply:
`StateDirectory = "redis";` will create `/var/lib/redis` which is not `/run/redis`. I am confused :thinking:

Reply:
This configuration from `mastodon` service - https://github.com/NixOS/nixpkgs/pull/117501/files#diff-157b1ef68573bbec951d6e551513a555e2d1ca7a161a68f1978b11d39a0bef1eR556-R559

Reply:
I also can see from this explanation why should have `RuntimeDirectoryMode = "0750";` so that /run/redis is group-readable. However this does not apply to `StateDirectoryMode = "0750";` as this changes permissions to `/var/lib/redis`. I don't think it needs to be readable to the redis group. It only needs to be accessible to the redis daemon.

Reply:
Example - required to be able to back up a file.

Reply:
Changed to "0700".